exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -975,67 +1685,236 @@ func (e *Exchange) GetHistoricCandlesExtended(ctx context.Context, pair currency
+		return nil, fmt.Errorf("%w LatestRateRequest", common.ErrEmptyParams)
+	}
+	var pairString string
+	if !r.Pair.IsEmpty() {
+		is, err := e.IsPerpetualFutureCurrency(r.Asset, r.Pair)
+		if err != nil {
+			return nil, err
+		} else if !is {
+			return nil, fmt.Errorf("%w %s %v", futures.ErrNotPerpetualFuture, r.Asset, r.Pair)
+		}
+		pairString = r.Pair.String()
+	}
+	contracts, err := e.GetFuturesHistoricalFundingRates(ctx, pairString, time.Time{}, time.Time{}, 0)
+	if err != nil {
+		return nil, err
+	}
+	timeChecked := time.Now()
+	resp := make([]fundingrate.LatestRateResponse, 0, len(contracts))
+	for _, fFundingRate := range contracts {
+		cp, err := currency.NewPairFromString(fFundingRate.Symbol)
+		if err != nil {
+			return nil, err
+		}
+		var isPerp bool
+		isPerp, err = e.IsPerpetualFutureCurrency(r.Asset, cp)
+		if err != nil {
+			return nil, err
+		} else if !isPerp {
+			continue
+		}
+		rate := fundingrate.LatestRateResponse{
+			Pair:     cp,
+			Exchange: e.Name,
+			Asset:    r.Asset,
+			LatestRate: fundingrate.Rate{
+				Time: fFundingRate.FundingRateSettleTime.Time(),
+				Rate: decimal.NewFromFloat(fFundingRate.FundingRate.Float64()),
+			},
+			TimeOfNextRate: fFundingRate.NextFundingTime.Time(),
+			TimeChecked:    timeChecked,
+		}
+		rate.PredictedUpcomingRate = fundingrate.Rate{
+			Time: fFundingRate.NextFundingTime.Time(),
+			Rate: decimal.NewFromFloat(fFundingRate.NextPredictedFundingRate.Float64()),
+		}
+		resp = append(resp, rate)
+	}
+	return resp, nil
+}
+
+// IsPerpetualFutureCurrency ensures a given asset and currency is a perpetual future
+func (e *Exchange) IsPerpetualFutureCurrency(a asset.Item, cp currency.Pair) (bool, error) {
+	return a == asset.Futures && strings.HasSuffix(cp.Quote.String(), "PERP"), nil
 }
 
 // UpdateOrderExecutionLimits updates order execution limits
-func (e *Exchange) UpdateOrderExecutionLimits(_ context.Context, _ asset.Item) error {
-	return common.ErrNotYetImplemented
+func (e *Exchange) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
+	if !e.SupportsAsset(a) {
+		return fmt.Errorf("%w: asset type: %v", asset.ErrNotSupported, a)
+	}
+	if a == asset.Spot {
+		instruments, err := e.GetSymbols(ctx)
+		if err != nil {
+			return err
+		}
+		l := make([]limits.MinMaxLevel, len(instruments))
+		for i, symbolDetail := range instruments {
+			cp, err := currency.NewPairFromString(symbolDetail.Symbol)
+			if err != nil {
+				return err
+			}
+			l[i] = limits.MinMaxLevel{
+				Key:                     key.NewExchangeAssetPair(e.Name, a, cp),
+				PriceStepIncrementSize:  symbolDetail.SymbolTradeLimit.PriceScale,
+				MinimumBaseAmount:       symbolDetail.SymbolTradeLimit.MinQuantity.Float64(),
+				MinimumQuoteAmount:      symbolDetail.SymbolTradeLimit.MinAmount.Float64(),
+				AmountStepIncrementSize: symbolDetail.SymbolTradeLimit.AmountScale,
+				QuoteStepIncrementSize:  symbolDetail.SymbolTradeLimit.QuantityScale,
```

Comment:
```suggestion
				PriceStepIncrementSize:  math.Pow(10, -symbolDetail.SymbolTradeLimit.PriceScale),
				MinimumBaseAmount:       symbolDetail.SymbolTradeLimit.MinQuantity.Float64(),
				MinimumQuoteAmount:      symbolDetail.SymbolTradeLimit.MinAmount.Float64(),
				AmountStepIncrementSize: math.Pow(10, -symbolDetail.SymbolTradeLimit.QuantityScale),
				QuoteStepIncrementSize:  math.Pow(10, -symbolDetail.SymbolTradeLimit.AmountScale),
```

Just so values are inline with the step increment so if base amount scale is 6 we can get 1e-6 as step size. Also quantity is the base amount and the amount == quote amount for some reason 🤷